drivers/pinmux/pinmux_mchp_xec.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <pinmux.h>
+#include <soc.h>
+
+static const u32_t valid_ctrl_masks[NUM_MCHP_GPIO_PORTS] = {
+	(MCHP_GPIO_PORT_A_BITMAP),
+	(MCHP_GPIO_PORT_B_BITMAP),
+	(MCHP_GPIO_PORT_C_BITMAP),
+	(MCHP_GPIO_PORT_D_BITMAP),
+	(MCHP_GPIO_PORT_E_BITMAP),
+	(MCHP_GPIO_PORT_F_BITMAP)
+};
+
+struct pinmux_xec_config {
+	__IO u32_t *pcr1_base;
+	u32_t port_num;
+};
+
+static int pinmux_xec_set(struct device *dev, u32_t pin, u32_t func)
+{
+	const struct pinmux_xec_config *config = dev->config->config_info;
+	__IO u32_t *current_pcr1;
+	u32_t pcr1 = 0;
+	u32_t mask = 0;
+
+	/* Validate pin number in terms of current port */
+	if ((valid_ctrl_masks[config->port_num] & BIT(pin)) == 0)
+		return -EINVAL;
+
+	mask |= MCHP_GPIO_CTRL_BUFT_MASK | MCHP_GPIO_CTRL_MUX_MASK;
+
+	/* Check for open drain/push_pull setting */
+	if (func & MCHP_GPIO_CTRL_BUFT_OPENDRAIN) {
+		pcr1 |= MCHP_GPIO_CTRL_BUFT_OPENDRAIN;
+	} else {
+		pcr1 |= MCHP_GPIO_CTRL_BUFT_PUSHPULL;
+	}
+
+	/* Parse mux mode */
+	pcr1 |= func & MCHP_GPIO_CTRL_MUX_MASK;
+
+	/* Figure out the pullup/pulldown configuration */
+	mask |= MCHP_GPIO_CTRL_PUD_MASK;
+
+	if (func & MCHP_GPIO_CTRL_PUD_PU) {
+		/* Enable the pull and select the pullup resistor. */
+		pcr1 |= MCHP_GPIO_CTRL_PUD_PU;
+	} else if (func & MCHP_GPIO_CTRL_PUD_PD) {
+		/* Enable the pull and select the pulldown resistor */
+		pcr1 |= MCHP_GPIO_CTRL_PUD_PD;
+	} else {
+		/* None : Pin tristates when no active driver is present
+		 * on the pin. This is the POR setting
+		 */
+		pcr1 |= MCHP_GPIO_CTRL_PUD_NONE;
+	}
+
+	/* Make sure gpio isrs are disabled */
+	pcr1 |= MCHP_GPIO_CTRL_IDET_DISABLE;
+	mask |= MCHP_GPIO_CTRL_IDET_DISABLE;
```

Comment:
MCHP GPIO interrupt detect is a multi-bit field in the GPIO control register.  The mask should be OR's with MCHP_GPIO_CTRL_IDET_MASK.

Reply:
I changed it.

Reply:
OK. Thanks